client/packages/electron/src/electron.ts

```diff
@@ -167,7 +171,25 @@ const start = (): void => {
       hardwareId,
     });
   });
+
+  window.webContents.on(
+    'did-fail-load',
+    (_event, _errorCode, errorDescription, validatedURL) => {
+      // not strictly necessary, done to prevent an infinite loop if the loadFile fails
+      if (hasLoadingError) return;
+
+      hasLoadingError = true;
+      const search = `error=Failed to load URL ${validatedURL} with error: ${errorDescription}`;
+      // loading './error.html' should work but doesn't, either as loadURL or loadFile
+      const filename = app.isPackaged
+        ? path.join(__dirname, '..', 'renderer', 'error.html')
+        : 'src/error.html';
+
+      window.loadFile(filename, { search });
```

Comment:
We could technically load from a sub route of rendered package (would mean error.html can be tsx file):
```diff
diff --git a/client/packages/common/src/index.ts b/client/packages/common/src/index.ts
index e71c77b53..71437ced8 100644
--- a/client/packages/common/src/index.ts
+++ b/client/packages/common/src/index.ts
@@ -28,6 +28,7 @@ export {
   useNavigate,
   useParams,
   BrowserRouter,
+  HashRouter,
   Routes,
   Route,
   Navigate,
diff --git a/client/packages/electron/src/electron.ts b/client/packages/electron/src/electron.ts
index d35ce196e..18541bed6 100644
--- a/client/packages/electron/src/electron.ts
+++ b/client/packages/electron/src/electron.ts
@@ -83,7 +83,7 @@ const getDebugHost = () => {
 // Can debug by opening chrome chrome://inspect and open inspect under 'devices'
 const START_URL = getDebugHost()
   ? `${getDebugHost()}/discovery`
-  : MAIN_WINDOW_WEBPACK_ENTRY;
+  : `${MAIN_WINDOW_WEBPACK_ENTRY}#/error`;
 
 // Handle creating/removing shortcuts on Windows when installing/uninstalling.
 if (require('electron-squirrel-startup')) {
diff --git a/client/packages/electron/src/home.tsx b/client/packages/electron/src/home.tsx
index 6a7a114a3..08e9ad98a 100644
--- a/client/packages/electron/src/home.tsx
+++ b/client/packages/electron/src/home.tsx
@@ -3,8 +3,11 @@ import { createRoot } from 'react-dom/client';
 import '@fontsource/inter/variable.css';
 import {
   AppThemeProvider,
+  HashRouter,
   IntlProvider,
   RandomLoader,
+  Route,
+  Routes,
   ServerDiscovery,
 } from '@openmsupply-client/common';
 import { Viewport } from '@openmsupply-client/host/src/components';
@@ -14,9 +17,17 @@ const ClientHomeScreen = () => (
     <IntlProvider isElectron={true}>
       <React.Suspense fallback={<RandomLoader />}>
         <AppThemeProvider>
-          <Viewport>
-            <ServerDiscovery />
-          </Viewport>
+          <HashRouter>
+            <Viewport>
+              <Routes>
+                <Route
+                  path="/error"
+                  element={<div>can be error compoenent here</div>}
+                />
+                <Route path="/" element={<ServerDiscovery />} />
+              </Routes>
+            </Viewport>
+          </HashRouter>
         </AppThemeProvider>
       </React.Suspense>
     </IntlProvider>
```